conda/environment.yml

```diff
@@ -3,27 +3,164 @@
 
 # !!! If you are developing the website locally use the existing lock files
 # for your platform (`conda create --name hackweek --file conda-osx-64.lock`)
+
+# Modified from https://github.com/CryoInTheCloud/hub-image/blob/main/environment.yml for UW Hackweek 2024
+
 name: hackweek
+
 channels:
   - conda-forge
+  - nodefaults
+
 dependencies:
-  - ipyleaflet
-  - jupyter-book<2
-  - jupyter-resource-usage
-  - jupyterhub-singleuser
-  - jupyterlab
-  - jupytext
-  - pip
-  - python
+  - python~=3.11.0
+  # Required until https://github.com/jupyterhub/repo2docker/pull/1196 is merged
+  - jupyterhub-singleuser~=4.0.1
+  # nbgitpuller is very helpful when distributing user content
+  - nbgitpuller~=1.2.0
+  # Specify major version of jupyterlab to use. Manually bump it when you want to upgrade
+  # jupyterlab version
+  - jupyterlab>=4.0,<4.1
+  # This provides memory usage stats on JupyterLab and classic notebook
+  - jupyter-resource-usage~=1.0.1
+  # https://syncthing.net/ provides a dropbox-like interface for syncing directories across users,
+  # hubs and machines. Used along with `jupyter-syncthing-proxy`.
+  # - jupyter-syncthing-proxy==1.0.3
+  # - syncthing~=1.22.1
+  # gh-scoped-creds allows users to securely push to GitHub from their repo.
+  # https://blog.jupyter.org/securely-pushing-to-github-from-a-jupyterhub-3ee42dfdc54f has
+  # some more information
+  # - gh-scoped-creds==4.1
+  # More git tools from github-cli
+  - gh~=2.32.1
+
+  # VS Code support
+  # - jupyter-vscode-proxy~=0.5
+  # - code-server~=4.16.1
+
+  # Jupyter tools
+  - ipython~=8.20.0
```

Comment:
Need to downgrade ipython here:

```suggestion
  - ipython~=8.17
```

so as to resolve this conflict with awscli=2.17:

```
Could not solve for environment specs
The following packages are incompatible
├─ awscli ~=2.17  is installable with the potential options
│  ├─ awscli [2.17.0|2.17.1|...|2.17.9] would require
│  │  └─ prompt_toolkit >=3.0.24,<3.0.39 , which can be installed;
│  ├─ awscli [2.17.0|2.17.1|...|2.17.9] would require
│  │  └─ python_abi 3.10.* *_cp310, which can be installed;
│  ├─ awscli [2.17.0|2.17.1|...|2.17.9] would require
│  │  └─ python_abi 3.8.* *_cp38, which can be installed;
│  ├─ awscli [2.17.0|2.17.1|...|2.17.9] would require
│  │  └─ pypy3.9 >=7.3.15 , which can be installed;
│  ├─ awscli [2.17.0|2.17.1|...|2.17.9] would require
│  │  └─ python_abi 3.9.* *_cp39, which can be installed;
│  └─ awscli [2.17.10|2.17.11|...|2.17.23] would require
│     └─ python_abi 3.12.* *_cp312, which can be installed;
├─ ipython ~=8.20  is not installable because there are no viable options
│  ├─ ipython [8.20.0|8.21.0|...|8.26.0] would require
│  │  └─ prompt-toolkit >=3.0.41,<3.1.0  but there are no viable options
│  │     ├─ prompt-toolkit 3.0.41 would require
│  │     │  └─ prompt_toolkit 3.0.41 , which conflicts with any installable versions previously reported;
│  │     ├─ prompt-toolkit 3.0.42 would require
│  │     │  └─ prompt_toolkit 3.0.42 , which conflicts with any installable versions previously reported;
│  │     ├─ prompt-toolkit 3.0.43 would require
│  │     │  └─ prompt_toolkit 3.0.43 , which conflicts with any installable versions previously reported;
│  │     ├─ prompt-toolkit 3.0.46 would require
│  │     │  └─ prompt_toolkit 3.0.46 , which conflicts with any installable versions previously reported;
│  │     └─ prompt-toolkit 3.0.47 would require
│  │        └─ prompt_toolkit 3.0.47 , which conflicts with any installable versions previously reported;
```

Because awscli=2.17 uses `prompt_toolkit >=3.0.24,<3.0.39`, we'll need an older version of ipython=8.17.2 that pins to `prompt_toolkit: '>=3.0.30,<3.1.0,!=3.0.37' until awscli relaxes the pin (e.g. at https://github.com/aws/aws-cli/pull/8734).